src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -88,7 +88,7 @@ internal static partial class ICodeDefinitionFactoryExtensions
 
             yield return CodeGenerationSymbolFactory.CreateConstructorSymbol(
                 attributes: default(ImmutableArray<AttributeData>),
-                accessibility: Accessibility.Public,
+                accessibility: containingTypeOpt.IsAbstractClass() ? Accessibility.Protected : Accessibility.Public,
```

Comment:
@CyrusNajmabadi, ```CreateBaseDelegatingConstructor``` used in ```AbstractGenerateTypeService```, I was not able to find out how generate abstract class with this refactoring. Could you please advice me how to do it?